### PR TITLE
IOTEDGE-1083 dynamic secret handling for examples

### PR DESCRIPTION
**Pull request by LimaEchoCharlie**

### CODE REVIEW

File: examples/secrets/secrets.go [hunk 1]
```diff
@@ -14,115 +14,181 @@
...
-	"\"y\":\"RT8XY1HCIso3HoUif9oBpwee2CBfjouqf2JjtFT6lKs\"," +
-	"\"d\":\"Fulx_Vjc-jRmYP_0K6TVFoIuK9MEYOgHCXRxCe3LZkY\"}]}"
+const ec256Test = `{"kty": "EC",
+	"kid": "Fol7IpdKeLZmzKtCEgi1LDhSIzM=",
+    "x": "N7MtObVf92FJTwYvY2ZvTVT3rgZp7a7XDtzT_9Rw7IA",
+	"y": "uxNmyoocPopYh4k1FCc41yuJZVohxlhMo3KTIJVTP3c",
+	"crv": "P-256",
+	"alg": "ES256",
+	"d": "w9rAMaNcP7cA0e5SECc4Tk1PDQEY66ml9y9-6E8fmR4",
+	"x5c": ["MIIBwjCCAWkCCQCw3GyPBTSiGzAJBgcqhkjOPQQBMGoxCzAJBgNVBAYTAlVLMRAwDgYDVQQIEwdCcmlzdG9sMRAwDgYDVQQHEwdCcmlzdG9sMRIwEAYDVQQKEwlGb3JnZVJvY2sxDzANBgNVBAsTBk9wZW5BTTESMBAGA1UEAxMJZXMyNTZ0ZXN0MB4XDTE3MDIwMzA5MzQ0NloXDTIwMTAzMDA5MzQ0NlowajELMAkGA1UEBhMCVUsxEDAOBgNVBAgTB0JyaXN0b2wxEDAOBgNVBAcTB0JyaXN0b2wxEjAQBgNVBAoTCUZvcmdlUm9jazEPMA0GA1UECxMGT3BlbkFNMRIwEAYDVQQDEwllczI1NnRlc3QwWTATBgcqhkjOPQIBBggqhkjOPQMBBwNCAAQ3sy05tV/3YUlPBi9jZm9NVPeuBmntrtcO3NP/1HDsgLsTZsqKHD6KWIeJNRQnONcriWVaIcZYTKNykyCVUz93MAkGByqGSM49BAEDSAAwRQIgZhTox7WpCb9krZMyHfgCzHwfu0FVqaJsO2Nl2ArhCX0CIQC5GgWD5jjCRlIWSEFSDo4DZgoQFXaQkJUSUbJZYpi9dA=="]
```
Comment (LimaEchoCharlie) [inline examples/secrets/secrets.go:45]: If I dynamically create the CA certificate, thing auth fails with:

`Path does not chain with any of the trust anchors`

**Comment by LimaEchoCharlie** [comment]
Go test fails because `*ecdsa.PrivateKey.Equal` was only introduced in Golang 1.15. Either, I could upgrade Go to 1.15 or write a bespoke equal function for the unit tests. I am happy to do either?

**Comment by JacoJooste** [comment]
> Go test fails because `*ecdsa.PrivateKey.Equal` was only introduced in Golang 1.15. Either, I could upgrade Go to 1.15 or write a bespoke equal function for the unit tests. I am happy to do either?

My vote would be for updating to Go 1.15.

File: examples/thing/cert-registration/main.go [hunk 2]
```diff
@@ -67,51 +44,59 @@ func loadKey() (crypto.Signer, error) {
...
-// To create your own certificate. Open a bash terminal:
-// Choose an ID for your thing:
-//     thingName=thingOne
-// Use openssl to create a certificate signing request and a certificate:
-//     openssl req -new -sha256 -subj "/CN=${thingName}" -key ./examples/resources/eckey1.key.pem -out "${thingName}.csr.pem"
-//     openssl x509 -req -CA ./examples/resources/es256test.cert.pem -CAkey ./examples/resources/es256test.key.pem -CAcreateserial -in "${thingName}.csr.pem" -out "${thingName}.cert.pem"
-// Ensure that the ID and the path to the certificate is passed to this example:
-//     go run github.com/ForgeRock/iot-edge/v7/examples/thing/cert-registration -name "${thingName}" -certfile "${thingName}.cert.pem"
 func certRegThing() (err error) {
+	var (
```
Comment (JacoJooste) [inline examples/thing/cert-registration/main.go:58]: We will have to change the docs to use the new options. Can you create a story for this please?

**Review by JacoJooste — APPROVED** [review]
This is great James.

**Comment by JacoJooste** [comment]
> Go test fails because `*ecdsa.PrivateKey.Equal` was only introduced in Golang 1.15. Either, I could upgrade Go to 1.15 or write a bespoke equal function for the unit tests. I am happy to do either?

We'll have to update the docs with the new version as well. Can you create a story for this please?

**Pull request merged**